### PR TITLE
Add AS3 template Parser

### PR DESCRIPTION
**Pull request by subbuv26**
This parser takes an AS3 template as input and return a map that has the parsed information related to Tenants, Applications ans Services.

Know Issues:

1. AS3 config is not being validated with respect to AS3 Schema.

### CODE REVIEW

File: pkg/appmanager/as3Manager.go [hunk 1]
```diff
@@ -17,10 +23,81 @@ func (appMgr *Manager) processUserDefinedAS3(template as3Template) {
 
 }
 
-// Story 2
-// Covert AS3 JSON in to Map for further processing
-func (appMgr *Manager) getAS3ObjectsFromTemplate(template as3Template) (as3Object, bool) {
-	return as3Object{}, true
+// getAS3ObjectFromTemplate gets an AS3 template as a input parameter.
+// It parses AS3 template and constructs and as3Object and
```
Comment (ssurenr) [inline pkg/appmanager/as3Manager.go]:
```suggestion
// It parses AS3 template, constructs an as3Object and
```

File: pkg/appmanager/as3Manager.go [hunk 2]
```diff
@@ -17,10 +23,81 @@ func (appMgr *Manager) processUserDefinedAS3(template as3Template) {
...
+// returns the same for further processing.
```
Comment (ssurenr) [inline pkg/appmanager/as3Manager.go]:
```suggestion
// returns it.
```

File: pkg/appmanager/as3Manager.go [hunk 3]
```diff
@@ -17,10 +23,81 @@ func (appMgr *Manager) processUserDefinedAS3(template as3Template) {
...
+func (appMgr *Manager) getAS3ObjectFromTemplate(
+	template as3Template,
+) (as3Object, bool) {
+	var tmpl interface{}
+	err := json.Unmarshal([]byte(template), &tmpl)
+	if err != nil {
+
```
Comment (ssurenr) [inline pkg/appmanager/as3Manager.go]: Let's remove the blank line

File: pkg/appmanager/as3Manager.go [hunk 4]
```diff
@@ -17,10 +23,81 @@ func (appMgr *Manager) processUserDefinedAS3(template as3Template) {
...
+		log.Error("Json unmarshal failed")
```
Comment (ssurenr) [inline pkg/appmanager/as3Manager.go]: Let's log the `err` too.

File: pkg/appmanager/as3Manager.go [hunk 5]
```diff
@@ -17,10 +23,79 @@ func (appMgr *Manager) processUserDefinedAS3(template as3Template) {
...
+				log.Debugf("No services declared for application: %s,"+
+					" tenant: %s\n", an, tn)
+			}
+		}
+		if len(as3[tenantName(tn)]) == 0 {
+			log.Debugf("No applications declared for tenant: %s\n", tn)
+		}
+	}
+	if len(as3) == 0 {
+		log.Error("No tenants declared in AS3 template")
```
Comment (trinaths) [inline pkg/appmanager/as3Manager.go]: log.Errorf - is the log error reporting

File: pkg/appmanager/as3Manager.go [hunk 6]
```diff
@@ -17,10 +23,79 @@ func (appMgr *Manager) processUserDefinedAS3(template as3Template) {
 
 }
 
-// Story 2
-// Covert AS3 JSON in to Map for further processing
```
Comment (trinaths) [inline pkg/appmanager/as3Manager.go]: can you add  more description. Like, Function arguments, return value and conditionals and finally a brief description

**Pull request merged**